src/context/LogContext/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, useCallback, useContext, useMemo } from "react";
+import useLogState from "./state";
+
+interface LogContextState {
+  logLines: string[];
+  lineCount: number;
+  ingestLines: (logs: string[]) => void;
+  clearLines: () => void;
+}
+const LogContext = createContext<LogContextState | null>(null);
+
+const useLogContext = () => {
+  const context = useContext(LogContext);
+  if (context === undefined) {
+    throw new Error("useLogContext must be used within a LogContextProvider");
+  }
+  return context as LogContextState;
+};
+
+const LogContextProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const { state, dispatch } = useLogState();
+
+  const ingestLines = useCallback(
+    (lines: string[]) => {
+      dispatch({ type: "INGEST_LOGS", logs: lines });
+    },
+    [dispatch]
+  );
+
+  const clearLines = useCallback(() => {
+    dispatch({ type: "CLEAR_LOGS" });
+  }, [dispatch]);
+
+  const memoizedContext = useMemo(
+    () => ({
+      logLines: state.logs,
+      lineCount: state.lineCount,
```

Comment:
In the future there may be parts of the state that are internal to the context that we wouldn't want to expose so it seemed to make sense to explicitly return values instead of returning everything.